TrafficCapture/coreUtilities/src/main/java/org/opensearch/migrations/tracing/ActiveContextTracker.java

```diff
@@ -0,0 +1,34 @@
+package org.opensearch.migrations.tracing;
+
+import java.util.Comparator;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentSkipListSet;
+import java.util.stream.Stream;
+
+public class ActiveContextTracker implements IContextTracker {
+    final ConcurrentSkipListSet<IScopedInstrumentationAttributes> orderedScopes;
+
+    public ActiveContextTracker() {
+        orderedScopes = makeScopeSkipList();
+    }
+
+    static ConcurrentSkipListSet<IScopedInstrumentationAttributes> makeScopeSkipList() {
+        return new ConcurrentSkipListSet<>(Comparator
+                .comparingLong(IWithStartTimeAndAttributes::getStartNanoTime)
+                .thenComparingInt(System::identityHashCode));
+    }
+
+    @Override
+    public void onContextCreated(IScopedInstrumentationAttributes scopedContext) {
+        orderedScopes.add(scopedContext);
+    }
+
+    @Override
+    public void onContextClosed(IScopedInstrumentationAttributes scopedContext) {
+        orderedScopes.remove(scopedContext);
+    }
+
+    public Stream<IScopedInstrumentationAttributes> getOldestActiveScopes() {
+        return orderedScopes.stream();
```

Comment:
I get that the oldest here is the ordering, however based on the method name i was expecting some sort of filtering on the active scopes

Reply:
Everything in this object will be active since the contexts are getting cleared out.  I don't mind the redundancy, but I'm also OK with 60 character names.  Let me know if you think it's clearer if it is shorter.